src/api/spec/policies/bs_request_policy_spec.rb

```diff
@@ -0,0 +1,113 @@
+require 'rails_helper'
+
+RSpec.describe BsRequestPolicy, type: :policy do
+  subject { BsRequestPolicy }
+
+  permissions :can_add_reviews? do
+    let(:user) { create(:confirmed_user, login: 'iggy') }
+    let(:author) { create(:confirmed_user, login: 'foo') }
+    let(:bs_request) { create(:bs_request_with_submit_action, state: 'review', creator: author) }
+    let(:review) { create(:review, state: 'new', by_user: user.login) }
+
+    context 'when the record is in state new' do
+      context 'and the user is a target maintainer' do
+        before do
+          allow(bs_request).to receive(:is_target_maintainer?).with(user).and_return(true)
+        end
+
+        it { expect(subject).to permit(user, bs_request) }
+      end
+
+      context 'and the user is not a target maintainer' do
+        before do
+          allow(bs_request).to receive(:is_target_maintainer?).with(user).and_return(false)
+        end
+
+        it { expect(subject).not_to permit(user, bs_request) }
+      end
+
+      context 'and there are open reviews present for the user' do
+        let(:review) { create(:review, state: 'new', by_user: user.login) }
+
+        before do
+          bs_request.reviews << review
+        end
+
+        it { expect(subject).to permit(user, bs_request) }
+      end
+
+      context 'and there are no open reviews present for the user' do
+        let(:review) { create(:review, state: 'accepted', by_user: user.login) }
+
+        before do
+          bs_request.reviews << review
+        end
+
+        it { expect(subject).not_to permit(user, bs_request) }
+      end
+
+      context 'and the user is the author of the record' do
+        it { expect(subject).to permit(author, bs_request) }
+      end
+
+      context 'and the user is not the author of the record' do
+        it { expect(subject).not_to permit(user, bs_request) }
+      end
+    end
+
+    context 'when the record is in state review' do
```

Comment:
This and `'when the record is in state new' ` is the same no?

Reply:
yes, somehow dropped changing the state of the request to `review` for this context. Its basically the same, but I thought its one of the conditions of the policy. But I will DRY it

Reply:
I moved the tests to a shared example and running it with the two states `review` and `new`